### PR TITLE
add preview workflow

### PR DESCRIPTION
**Pull request by maelle**

### CODE REVIEW

File: .github/workflows/pkgdown.yaml [hunk 1]
```diff
@@ -45,7 +51,37 @@ jobs:
...
+          Rscript -e 'pkgdown::build_home(override=list(destination="cosmo", template = list(params=list( bootswatch = "cosmo"))))'
+
+      - name: Move files
+        if: ${{ github.event_name == 'pull_request' && github.event.pull_request.head.repo.fork == 'true' }}
+        run: |
+          cp -r ./cosmo/dev ./docs/dev/cosmo
+
+
+      - name: Deploy to Netlify
+        if: ${{ github.event_name == 'pull_request' && github.event.pull_request.head.repo.fork == 'true' }}
```
Comment (maelle) [inline .github/workflows/pkgdown.yaml:73]: fork will eventually be false but I want to test it works... from my fork.

File: .github/workflows/pkgdown.yaml [hunk 2]
```diff
@@ -45,7 +53,37 @@ jobs:
...
         run: |
           git config --local user.email "actions@github.com"
           git config --local user.name "GitHub Actions"
           Rscript -e 'pkgdown::deploy_to_branch(new_process = FALSE)'
+
+      - name: Create website and another page
+        if: contains(env.isPush, 'false')
+        run: |
+          Rscript -e 'pkgdown::build_site()'
+          Rscript -e 'pkgdown::build_home(override=list(destination="cosmo", template = list(params=list( bootswatch = "cosmo"))))'
```
Comment (maelle) [inline .github/workflows/pkgdown.yaml:66]: I wanted to quickly see a Bootswatch theme still works, but maybe not needed here.

File: .github/workflows/pkgdown.yaml [hunk 3]
```diff
@@ -45,7 +53,37 @@ jobs:
...
 
       - name: Deploy package
+        if: contains(env.isPush, 'true')
         run: |
           git config --local user.email "actions@github.com"
           git config --local user.name "GitHub Actions"
           Rscript -e 'pkgdown::deploy_to_branch(new_process = FALSE)'
+
+      - name: Create website and another page
+        if: contains(env.isPush, 'false')
```
Comment (hadley) [inline .github/workflows/pkgdown.yaml:63]: Can you please use the run + shell technique? (See the install dependencies step above)

File: .github/workflows/pkgdown.yaml (hunk 2 above)
Comment (hadley) [reply .github/workflows/pkgdown.yaml:66]: Yeah, lets test that separately

File: .github/workflows/pkgdown.yaml [hunk 4]
```diff
@@ -45,7 +53,37 @@ jobs:
...
+          Rscript -e 'pkgdown::build_home(override=list(destination="cosmo", template = list(params=list( bootswatch = "cosmo"))))'
+
+      - name: Move files
+        if: contains(env.isPush, 'false')
+        run: |
+          cp -r ./cosmo/dev ./docs/dev/cosmo
+
+
+      - name: Deploy to Netlify
+        if: contains(env.isPush, 'false') && contains(env.isExtPR, 'false')
```
Comment (hadley) [inline .github/workflows/pkgdown.yaml:75]: Where is `env.isExtPR` set?

Comment (maelle) [reply .github/workflows/pkgdown.yaml:75]: forgot to remove it, it was from before the whole workflow was skipped for external PRs.

File: .github/workflows/pkgdown.yaml (hunk 3 above)
Comment (maelle) [reply .github/workflows/pkgdown.yaml:63]: done

**Comment by hadley** [comment]
Why isn't the preview generated for this PR?

**Comment by maelle** [comment]
because it comes from my fork

**Comment by hadley** [comment]
Ah ok, but you'll submit future PRs from this repo, right?

**Comment by maelle** [comment]
yes!

**Pull request merged**